public/app/features/manage-dashboards/state/actions.ts

```diff
@@ -45,17 +46,77 @@ export function fetchGcomDashboard(id: string): ThunkResult<void> {
 
 export function importDashboardJson(dashboard: any): ThunkResult<void> {
   return async (dispatch) => {
+    await dispatch(processElements(dashboard));
+    await dispatch(processJsonDashboard(dashboard));
+    dispatch(processInputs());
+  };
+}
+
+const isDataSourceBeingConsumed = (
+  datasource: string | { uid?: string } | undefined | null,
+  datasourceName: string
+): boolean => {
+  if (!datasource) {
+    return false;
+  }
+
+  return typeof datasource === 'string'
+    ? getDataSourceRefNameFromPlaceholder(datasource) === datasourceName
+    : getDataSourceRefNameFromPlaceholder(datasource?.uid) === datasourceName;
```

Comment:
unfortunately, with gcom request I receive a datasource that is just a string (the datasourceId) and when I export it, I receive a datasource that is an object